src/connector/src/sink/postgres.rs

```diff
@@ -0,0 +1,625 @@
+                "host={} port={} user={} password={} dbname={}",
+                config.host, config.port, config.user, config.password, config.database
+            );
+            let (client, connection) =
+                tokio_postgres::connect(&connection_string, tokio_postgres::NoTls)
+                    .await
+                    .context("Failed to connect to Postgres for Sinking")?;
+            tokio::spawn(async move {
+                if let Err(error) = connection.await {
+                    tracing::error!(error = %error.as_report(), "postgres sink connection error");
+                }
+            });
+            client
+        };
+
+        let insert_statement = {
+            let insert_types = schema
+                .fields()
+                .iter()
+                .map(|field| field.data_type().to_pg_type())
+                .collect_vec();
+            let insert_sql = create_insert_sql(&schema, &config.table);
+            client
+                .prepare_typed(&insert_sql, &insert_types)
+                .await
+                .context("Failed to prepare insert statement")?
+        };
+
+        let delete_statement = if is_append_only {
+            None
+        } else {
+            let delete_types = pk_indices
+                .iter()
+                .map(|i| schema.fields()[*i].data_type().to_pg_type())
+                .collect_vec();
+            let delete_sql = create_delete_sql(&schema, &config.table, &pk_indices);
+            Some(
+                client
+                    .prepare_typed(&delete_sql, &delete_types)
+                    .await
+                    .context("Failed to prepare delete statement")?,
+            )
+        };
+
+        let merge_statement = if is_append_only {
+            None
+        } else {
+            let merge_types = schema
+                .fields
+                .iter()
+                .map(|field| field.data_type().to_pg_type())
+                .collect_vec();
+            let merge_sql = create_merge_sql(&schema, &config.table, &pk_indices);
+            Some(
+                client
+                    .prepare_typed(&merge_sql, &merge_types)
+                    .await
+                    .context("Failed to prepare merge statement")?,
+            )
+        };
+
+        let writer = Self {
+            config,
+            pk_indices,
+            is_append_only,
+            client,
+            buffer: Buffer::new(),
+            insert_statement,
+            delete_statement,
+            merge_statement,
+        };
+        Ok(writer)
+    }
+
+    async fn flush(&mut self) -> Result<()> {
+        if self.is_append_only {
+            for chunk in self.buffer.drain() {
+                for (op, row) in chunk.rows() {
+                    match op {
+                        Op::Insert => {
+                            self.client
+                                .execute_raw(&self.insert_statement, row.iter())
```

Comment:
It seems we insert each row of a chunk one by one, the performance could be sub-optimal compared to bulk insert. e.g. `INSERT INTO t values (a1,b1,c2),(a2,b2,c2),...`

Reply:
Optimize it later when I'm doing benchmarking. The entire chunk may have interleaving of `insert`, `delete`, `update`. I'm not sure if there will be a clear performance benefit.

Does jdbc sink batch insert?

Reply:
> Optimize it later when I'm doing benchmarking. The entire chunk may have interleaving of `insert`, `delete`, `update`. I'm not sure if there will be a clear performance benefit.
> 
> Does jdbc sink batch insert?

We have some [improvements](https://github.com/risingwavelabs/risingwave/pull/17174) to supported the batch DML to downstream.

Reply:
https://github.com/risingwavelabs/risingwave/pull/12250